log4j-core/src/main/java/org/apache/logging/log4j/core/util/internal/instant/InstantPatternDynamicFormatter.java

```diff
@@ -711,23 +488,114 @@ public void formatTo(final StringBuilder buffer, final Instant instant) {
                 }
             };
         }
+
+        @Override
+        @Nullable
+        PatternFormatterFactory tryMerge(PatternFormatterFactory other, ChronoUnit thresholdPrecision) {
+            // We always merge consecutive static pattern factories
+            if (other instanceof StaticPatternFormatterFactory) {
+                final StaticPatternFormatterFactory otherStatic = (StaticPatternFormatterFactory) other;
+                return new StaticPatternFormatterFactory(this.literal + otherStatic.literal);
+            }
+            // We also merge a static pattern factory with a DTF factory
+            if (other instanceof DateTimeFormatterPatternFormatterFactory) {
+                final DateTimeFormatterPatternFormatterFactory otherDtf =
+                        (DateTimeFormatterPatternFormatterFactory) other;
+                return new DateTimeFormatterPatternFormatterFactory(
+                        this.pattern + otherDtf.pattern, otherDtf.precision);
+            }
+            return super.tryMerge(other, thresholdPrecision);
+        }
+
+        static String escapeLiteral(String literal) {
+            StringBuilder sb = new StringBuilder(literal.length() + 2);
+            boolean inSingleQuotes = false;
+            for (int i = 0; i < literal.length(); i++) {
+                char c = literal.charAt(i);
+                if (c == '\'') {
+                    if (inSingleQuotes) {
+                        sb.append("'");
+                    }
+                    inSingleQuotes = false;
+                    sb.append("''");
+                } else {
+                    if (!inSingleQuotes) {
+                        sb.append("'");
+                    }
+                    inSingleQuotes = true;
+                    sb.append(c);
+                }
+            }
+            if (inSingleQuotes) {
+                sb.append("'");
+            }
+            return sb.toString();
+        }
     }
 
-    static final class DynamicPatternSequence extends PatternSequence {
+    /**
+     * Formats the pattern using {@link DateTimeFormatter}.
+     */
+    static final class DateTimeFormatterPatternFormatterFactory extends PatternFormatterFactory {
```

Comment:
*Nit:* I prefer

1. renaming this from `DateTimeFormatter...` to `Dynamic...` (since the usage of `DTF` has nothing to do with its function, which is being a dynamic date & time pattern ~sequence~ formatter factory)
2. extending `SecondPattern...` from this `Dynamic...`

Reply:
I renamed it to `Dynamic`, but I'd rather keep `SecondPatternSequence` separate.